test/functional/ui/syntax_conceal_spec.lua

```diff
@@ -0,0 +1,402 @@
+            ~                                                    |
+            ~                                                    |
+                                                                 |
+          ]])
+        end)
+
+        it('and conceal its start tag and end tag.', function() 
+          execute("syn region rText matchgroup=rMatch start='<r>' end='</r>' concealends cchar=-")  
+          screen:expect([[
+            {1:-} a region of text {1:-}                                 |
+            {1:-} a region of text {1:-}                                 |
+            ^                                                     |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+                                                                 |
+          ]])
+        end)
+      end)
+      
+      it('that are nested and conceal the nested region\'s start and end tags.', function()
+        execute("syn region rText contains=rText matchgroup=rMatch start='<r>' end='</r>' concealends cchar=-")  
+        insert("<r> A region with <r> a nested <r> nested region.</r> </r> </r>\n")
+        screen:expect([[
+          {1:-} a region of text {1:-}                                 |
+          {1:-} a region of text {1:-}                                 |
+          {1:-} A region with {1:-} a nested {1:-} nested region.{1:-}         |
+           {1:-} {1:-}                                                 |
+          ^                                                     |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+                                                               |
+        ]])
+      end)
+    end) -- Region
+
+    describe("a region of text", function()
+      local syn_region_i_text = "syn region iText start='<i>' end='</i>' cchar=*"
+      local conceal_on = "syntax conceal on"
```

Comment:
I think these, at least `conceal_on` and `conceal_off`, should be eliminated and the full command just put inside `execute()`. It is clearer if the command is just entered where it actually is used.